src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -77,92 +78,102 @@ public static IEnumerable<VisualStudioInstance> GetInstances(string workingDirec
             }
         }
 
+        enum hostfxr_resolve_sdk2_flags_t
+        {
+            disallow_prerelease = 0x1,
+        };
+
+        enum hostfxr_resolve_sdk2_result_key_t
+        {
+            resolved_sdk_dir = 0,
+            global_json_path = 1,
+        };
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl, CharSet = CharSet.Auto)]
+        private delegate void hostfxr_resolve_sdk2_result_fn(
+                hostfxr_resolve_sdk2_result_key_t key,
+                string value);
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl, CharSet = CharSet.Auto)]
+        private delegate void hostfxr_get_available_sdks_result_fn(
+                hostfxr_resolve_sdk2_result_key_t key,
+                [MarshalAs(UnmanagedType.LPArray, SizeParamIndex = 0)]
+                string[] value);
+
+        [DllImport("hostfxr", CharSet = CharSet.Auto, ExactSpelling = true, CallingConvention = CallingConvention.Cdecl)]
+        private static extern int hostfxr_resolve_sdk2(
+            string exe_dir,
+            string working_dir,
+            hostfxr_resolve_sdk2_flags_t flags,
+            hostfxr_resolve_sdk2_result_fn result);
+
+        [DllImport("hostfxr", CharSet = CharSet.Auto, ExactSpelling = true, CallingConvention = CallingConvention.Cdecl)]
+        private static extern int hostfxr_get_available_sdks(string exe_dir, hostfxr_get_available_sdks_result_fn result);
+
         private static IEnumerable<string> GetDotNetBasePaths(string workingDirectory)
         {
-            const string DOTNET_CLI_UI_LANGUAGE = nameof(DOTNET_CLI_UI_LANGUAGE);
-            
-            Process process;
-            var lines = new List<string>();
-            try
+            string dotnetPath = File.Exists("/usr/share/dotnet/dotnet") ? "/usr/share/dotnet" :
+            File.Exists($"/home/{Environment.GetEnvironmentVariable("USER")}/share/dotnet/dotnet") ? $"/home/{Environment.GetEnvironmentVariable("USER")}/share/dotnet" :
+            File.Exists("dotnet") ? "." :
+            !string.IsNullOrEmpty(Environment.GetEnvironmentVariable("DOTNET_HOME")) ? Environment.GetEnvironmentVariable("DOTNET_HOME") :
+            null;
```

Comment:
I asked about the recommended way to find dotnet. I believe your test got different results because the old version skipped to just checking your PATH, whereas this new version looks at several other places that seem more reasonable (if dotnet is there) than PATH. Those other places don't know about this random SDK you just installed but do know about the SDKs you installed the normal way. This is not a case in which I think we should be optimizing for bad customers.

Reply:
I think it makes sense to replicate "the build that you would get if you typed `dotnet build` instead of `toolThatUsesLocator`". That would respect `PATH` and multi-level lookup and friends and would not bypass them to find a machine-installed SDK.

Reply:
> That would respect PATH and multi-level lookup and friends and would not bypass them to find a machine-installed SDK.

That would be a good integration test 👍